File: src/main/java/musiccube/entities/Rate.java
package musiccube.entities;

import javax.persistence.*;

@Entity
@Table(name = "rate")
public class Rate {

    @Id
    @Column(name = "rate_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne
    @JoinColumn(name = "user_id", referencedColumnName = "user_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "song_id", referencedColumnName = "song_id")
    private Song song;

    @ManyToOne
    @JoinColumn(name = "id")
    private Album album;

    @ManyToOne
    @JoinColumn(name = "band_id", referencedColumnName = "band_id")
    private Band band;

    @ManyToOne
    @JoinColumn(name = "person_id", referencedColumnName = "person_id")
    private Artist artist;

    @Column(name = "rate")
    private int rate;

    public Rate() {
    }

    private Rate(User user, int rate) {
        this.user = user;
        this.rate = rate;
    }

    public Rate(User user, Song song, int rate) {
        this(user, rate);
        this.song = song;
    }

    public Rate(User user, Album album, int rate) {
        this(user, rate);
        this.album = album;
    }

    public Rate(User user, Band band, int rate) {
        this(user, rate);
        this.band = band;
    }

    public Rate(User user, Artist artist, int rate) {
        this(user, rate);
        this.artist = artist;
    }


    public int getId() {
        return id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Song getSong() {
        return song;
    }

    public void setSong(Song song) {
        this.song = song;
    }

    public int getRate() {
        return rate;
    }

    public void setRate(int rate) {
        this.rate = rate;
    }

    public Album getAlbum() {
        return album;
    }

    public void setAlbum(Album album) {
        this.album = album;
    }

    public Band getBand() {
        return band;
    }

    public void setBand(Band band) {
        this.band = band;
    }

    public Artist getArtist() {
        return artist;
    }

    public void setArtist(Artist artist) {
        this.artist = artist;
    }
}

File: Frontend/src/app/Class/Rate.ts
import { Users } from './Users';
import { Song } from './Song';
import { Album } from './Album';
import { Band } from './Band';
import { Artist } from './Artist';

export class Rate {

    private id: number;
    private user: Users;
    private song: Song;
    private album: Album;
    private band: Band;
    private artist: Artist;
    private rate: number;

    constructor(obj?: any) {
        this.id = (obj && obj.id) || 0;
        this.user = (obj && obj.user) || null;
        this.song = (obj && obj.song) || null;
        this.album = (obj && obj.album) || null;
        this.band = (obj && obj.band) || null;
        this.artist = (obj && obj.artist) || null;
        this.rate = (obj && obj.rate) || 0;
    }

    getId(): number {
        return this.id;
    }

    getUser(): Users {
        return this.user;
    }

    getSong(): Song {
        return this.song;
    }

    getAlbum(): Album {
        return this.album;
    }

    getBand(): Band {
        return this.band;
    }

    getArtist(): Artist {
        return this.artist;
    }

    getRate(): number {
        return this.rate;
    }

    setUser(user: Users) {
        this.user = user;
    }
    
    setSong(song: Song) {
        this.song = song;
    }

    setAlbum(album: Album) {
        this.album = album;
    }

    setBand(band: Band) {
        this.band = band;
    }

    setArtist(artist: Artist) {
        this.artist = artist;
    }
    
    setRate(rate: number) {
        this.rate = rate;
    }
}
